### PR TITLE
fix: softmax and bn+scale fusion

### PR DESCRIPTION
**Pull request by i8run**
## What changes were proposed in this pull request?

This patch add two features about the SoftMax and BN + Scale fusion.

## How was this patch tested?

Unit tests and tests by manual.

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala [hunk 1]
```diff
@@ -394,14 +394,15 @@ class DnnGraph(
...
-      for (j <- 0 to 3) {
+      for (j <- 0 to 4) {
         var i = forwardExecution.length - 1
         while (i >= 0) {
           if (j == 0) Fusion.fuseModule(forwardExecution(i))
           // we should do this before sum fusion, because it will change the structure of graph
           if (j == 1) Fusion.setNegativeInputOfConv(forwardExecution(i))
           if (j == 2) Fusion.fuseCAdd(forwardExecution(i))
           if (j == 3) Fusion.setScalesPrevousJoinTable(forwardExecution(i))
+          if (j == 4) Fusion.fuseModule(forwardExecution(i))
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala]: Why fuse models again?

Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala]: You are right. I'll fix it.

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalizationSpec.scala [hunk 2]
```diff
@@ -219,6 +219,32 @@ class SpatialBatchNormalizationSpec extends FlatSpec with Matchers with BeforeAn
...
+
+    bn.parameters()._1(0).copy(scale.parameters()._1(0))
+    bn.parameters()._1(1).copy(scale.parameters()._1(1))
+
+    scale.parameters()._1(0).fill(1.0f)
+    scale.parameters()._1(1).fill(0.0f)
+
+    val output2 = bn.forward(input)
+
+    println("")
```
Comment (wzhongyuan) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalizationSpec.scala]: delete it ?

**Comment by i8run** [comment]
Jenkins has passed. http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/984/

**Pull request merged**